examples/plot_compare_reduction.py

```diff
@@ -1,30 +1,47 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 # -*- coding: utf-8 -*-
 """
-=================================================================
-Selecting dimensionality reduction with Pipeline and GridSearchCV
-=================================================================
+=======================================================================
+Selecting dimensionality reduction with Pipeline, CachedPipeline, and \
+GridSearchCV
+=======================================================================
 
 This example constructs a pipeline that does dimensionality
 reduction followed by prediction with a support vector
-classifier. It demonstrates the use of GridSearchCV and
-Pipeline to optimize over different classes of estimators in a
-single CV run -- unsupervised PCA and NMF dimensionality
+classifier. It demonstrates the use of ``GridSearchCV`` and
+``Pipeline`` to optimize over different classes of estimators in a
+single CV run -- unsupervised ``PCA`` and ``NMF`` dimensionality
 reductions are compared to univariate feature selection during
 the grid search.
+
+Additionally, ``Pipeline`` can be exchanged with ``CachedPipeline``
+to memoize the transformers within the pipeline, avoiding to fit
+again the same transformers over and over.
+
+Note that the use of ``CachedPipeline`` becomes interesting when the
+fitting of a transformer is costly.
 """
+
+###############################################################################
+# Illustration of ``Pipeline`` and ``GridSearchCV``
+###############################################################################
+# This section illustrates the use of a ``Pipeline`` with
+# ``GridSearchCV``
+
 # Authors: Robert McGibbon, Joel Nothman
 
 from __future__ import print_function, division
 
+from tempfile import mkdtemp
 import numpy as np
 import matplotlib.pyplot as plt
 from sklearn.datasets import load_digits
 from sklearn.model_selection import GridSearchCV
-from sklearn.pipeline import Pipeline
+from sklearn.pipeline import Pipeline, CachedPipeline
```

Comment:
I would import this only later: trying to separate a bit the two parts of the example.

We have to keep in mind that every added piece of information to an example makes it harder to understand.